lib/knife/changelog/changelog.rb

```diff
@@ -185,6 +191,11 @@ def detect_cur_revision(name, rev, git)
       unless git.revision_exists?(rev)
         prefixed_rev = 'v' + rev
         return prefixed_rev if git.revision_exists?(prefixed_rev)
+        if rev.split('.')[-1] == '0' # It is the first rev for a major/minor, might be missing
```

Comment:
Can you please wrap this code in a method and add a TODO saying we need to remove it after the issue is solved with the versioning of the java cookbook? For me, it's a denormalization we need to avoid.

Reply:
I can, though I don't see the value. The method is not complex and what is being done is pretty obvious. As for the TODO, I actually disagree - the fix should stay. This change is not just for today's state of the Java cookbook, it is a reasonable compromise solution that @kamaradclimber proposed for dealing with missing revisions.

Reply:
If you're two against me, then I give up :).